RIOT-OS/RIOT
# cpu: Rename CPU_ARCH to CPU_CORE

### Contribution description
Currently cortexm and mips CPUs are using the variable `CPU_ARCH` to refer to the CPU core identifier. Even the documentation of this variable indicates that:
https://github.com/RIOT-OS/RIOT/blob/58cd1265178a767479850236f2b78598f0d217b2/makefiles/vars.inc.mk#L15

A naming discussion started in #14176 regarding our classification system for our CPUs. It makes sense to structure the names using the correct terms. This PR introduces `CPU_CORE` as a replacement for `CPU_ARCH`, and assigns the correct architecture value to cortexm and mips CPUs.

### Testing procedure
- Use `make dependency-debug` to check that the correct `CPU_CORE` and `CPU_ARCH` is assign to the boards

### Issues/PRs references
Somehow related to #14176
